### PR TITLE
Fix potential server-side timeout

### PR DESCRIPTION
**Pull request by RoiEXLab**
Follow up to #6104 should help debug this issue

## Functional Changes

[] New map or map update
[] New Feature
[] Feature update or enhancement
[] Feature Removal
[] Code Cleanup or refactor
[] Configuration Change
[x] Problem fix:  
[] Other:   

## Testing

I verified I could start a local networked game without any problems

### CODE REVIEW

File: game-core/src/main/java/games/strategy/engine/message/RemoteMethodCall.java [hunk 1]
```diff
@@ -94,6 +95,12 @@ public String getMethodName() {
   }
 
   private static String[] classesToString(final Class<?>[] classes, final Object[] args) {
+    if (args != null && classes.length != args.length) {
+      throw new IllegalArgumentException(
+          String.format(
+              "Classes and args arrays diff in length: %s, %s",
+              Arrays.toString(classes), Arrays.toString(args)));
```
Comment (RoiEXLab) [inline game-core/src/main/java/games/strategy/engine/message/RemoteMethodCall.java:102]: This is where I ran into #6136 FYI

File: game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java [hunk 2]
```diff
@@ -117,7 +119,12 @@ private RemoteMethodCallResults invokeAndWaitRemote(final RemoteMethodCall remot
     final Invoke invoke = new HubInvoke(methodCallId, true, remoteCall);
     send(invoke, messenger.getServerNode());
 
-    Interruptibles.await(latch);
+    if (!Interruptibles.awaitResult(() -> latch.await(1, TimeUnit.MINUTES)).result.orElse(false)) {
```
Comment (DanVanAtta) [inline game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java:122]: Is 1 minute potentially an over-generous timeout?

Have you simulated this error by any chance? Did it show the option to submit an error report (I'm hoping yes)

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java:122]: I haven't tested if there is an error dialog, but given that a couple of lines below a similar exception is thrown, hopes are that we'll get an error dialog.
But I can test this tomorrow.

Comment (RoiEXLab) [reply game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java:122]: And the 1 minute is arbitrary, open for suggestions

Comment (DanVanAtta) [reply game-core/src/main/java/games/strategy/engine/message/unifiedmessenger/UnifiedMessenger.java:122]: 20 seconds?

**Comment by RoiEXLab** [comment]
@DanVanAtta I changed it so it's just 20 seconds now, however I had to change some of the code because I realized it could lead to a memory leak otherwise.

Also I tested if we get a dialog and we don't. I'm not quite sure why though, because the exception is properly logged with level SEVERE:
```
Apr. 06, 2020 10:26:40 VORM. org.triplea.game.client.HeadedGameRunner lambda$initializeClientSettingAndLogging$0
SCHWERWIEGEND: Server timed out while waiting for result of method joinChat for remote _ChatRemote_games.strategy.engine.framework.ui.ServerStartup.CHAT_NAME with id dbdaddc9-6648-445e-8a8a-86f746d1704e
java.lang.IllegalStateException: Server timed out while waiting for result of method joinChat for remote _ChatRemote_games.strategy.engine.framework.ui.ServerStartup.CHAT_NAME with id dbdaddc9-6648-445e-8a8a-86f746d1704e
	at games.strategy.engine.message.unifiedmessenger.UnifiedMessenger.invokeAndWaitRemote(UnifiedMessenger.java:125)
	at games.strategy.engine.message.unifiedmessenger.UnifiedMessenger.invokeAndWait(UnifiedMessenger.java:91)
	at games.strategy.engine.message.UnifiedInvocationHandler.invoke(UnifiedInvocationHandler.java:48)
	at com.sun.proxy.$Proxy4.joinChat(Unknown Source)
	at games.strategy.engine.chat.MessengersChatTransmitter.connect(MessengersChatTransmitter.java:74)
	at games.strategy.engine.chat.Chat.<init>(Chat.java:53)
	at games.strategy.engine.chat.ChatPanel.newChatPanel(ChatPanel.java:49)
	at games.strategy.engine.framework.startup.mc.ClientModel.createClientMessenger(ClientModel.java:238)
	at games.strategy.engine.framework.startup.mc.SetupPanelModel.showClient(SetupPanelModel.java:101)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by DanVanAtta** [comment]
@RoiEXLab found a missing initialize call to error dialog: https://github.com/triplea-game/triplea/pull/6152

**Pull request merged**